internal/hcs/system.go

```diff
@@ -358,16 +353,20 @@ func (computeSystem *System) waitBackground() {
 
 // Wait synchronously waits for the compute system to shutdown or terminate. If
 // the compute system has already exited returns the previous error (if any).
-func (computeSystem *System) Wait() (err error) {
+func (computeSystem *System) Wait(ctx context.Context) (err error) {
```

Comment:
I think we should only return `err` if `<-computeSystem.waitBlock` finishes. At a minimum we should return an `unwrapped` error for `ctx.Err()`.

```go
ctx, cancel := context.WithDeadline(context.Background(), 30 * time.Second)
defer cancel()
err := sys.Wait(ctx)
if err != nil {
    if err == context.DeadlineExceeded {
        // Return a timeout error
    } else if hcs.IsSomeHcsErr(err) {
        // Do something to correct for error
    }
}
```

So this code just goes to:
```go
select {
case err := <-computeSystem.waitBlock:
    if err != nil {
        return makeSystemError(computeSystem, "Wait", "", err, nil)
    }
case <-ctx.Done():
}
return nil
```